src/table.js

```diff
@@ -488,30 +489,45 @@ Table.prototype.createReadStream = function(options) {
   if (options.limit) {
     reqOpts.rowsLimit = options.limit;
   }
-
+  const chunkFormatter = new ChunkFormatter();
   var stream = pumpify.obj([
     this.requestStream(grpcOpts, reqOpts),
-    through.obj(function(data, enc, next) {
-      var throughStream = this;
-      var rows = Row.formatChunks_(data.chunks, {
-        decode: options.decode,
-      });
-
-      rows.forEach(function(rowData) {
-        if (stream._ended) {
-          return;
+    through.obj(
+      function(data, enc, next) {
+        var throughStream = this;
+        try {
+          chunkFormatter.formatChunks(
+            data.chunks,
+            {
+              decode: options.decode,
+            },
+            (err, rowData) => {
+              if (stream._ended) {
+                throw new Error('stream ended');
```

Comment:
It appears that you throw this error to "break" out of the loop in formatChunks, and catch the error on line 515. This doesn't feel like idiomatic javascript, can you instead return false in this loop and that will signal to `formatChunks` that you want to terminate early? Lodash follows a similar pattern https://lodash.com/docs/4.17.4#forEach